File: fix_all.js
const fs = require('fs');
const path = require('path');

function slugify(text) {
    return text.toString().toLowerCase()
        .replace(/\s+/g, '-')           // Replace spaces with -
        .replace(/%20/g, '-')           // Replace %20 with -
        .replace(/[^\w\-]+/g, '')       // Remove all non-word chars except hyphen (wait, what about extension?)
        .replace(/\-\-+/g, '-')         // Replace multiple - with single -
        .replace(/^-+/, '')             // Trim - from start of text
        .replace(/-+$/, '');            // Trim - from end of text
}

function cleanFilename(filename) {
    const ext = path.extname(filename).toLowerCase();
    const name = path.basename(filename, path.extname(filename));
    
    // special handling: just replace spaces with hyphens and lowercase
    let newName = name.toLowerCase().replace(/[\s%20]+/g, '-').replace(/\-\-+/g, '-');
    return newName + ext;
}

const dirsToRename = {
    'canecas site': 'canecas-site',
    'Placas de MDF': 'placas-de-mdf'
};

const mapOldToNew = {}; // store mapping for file contents replacement

for (const [oldDir, newDir] of Object.entries(dirsToRename)) {
    if (fs.existsSync(oldDir)) {
        console.log(`Renaming folder: "${oldDir}" -> "${newDir}"`);
        fs.renameSync(oldDir, newDir);
    }
    
    // Now process files in the new dir
    if (fs.existsSync(newDir)) {
        const files = fs.readdirSync(newDir);
        for (const file of files) {
            const oldFilePath = path.join(newDir, file);
            const stats = fs.statSync(oldFilePath);
            if (stats.isFile()) {
                const newFilename = cleanFilename(file);
                if (newFilename !== file) {
                    const newFilePath = path.join(newDir, newFilename);
                    fs.renameSync(oldFilePath, newFilePath);
                    console.log(`Renamed file: "${file}" -> "${newFilename}"`);
                }
                
                // Keep track of the old full path mapping (olddir/oldfile -> newdir/newfile)
                // We need to account for what might be in products-data.js
                // products-data.js might have: "canecas site/Stitich cabo rosa 1 .jpg"
                const oldUrlPath = `${oldDir}/${file}`;
                const newUrlPath = `${newDir}/${newFilename}`;
                mapOldToNew[oldUrlPath.toLowerCase()] = newUrlPath;
            }
        }
    }
}

// Now replace in products-data.js and other files
const filesToProcess = ['products-data.js', 'index.html']; // You can add more if needed
for (const rootFile of fs.readdirSync('.')) {
    if ((rootFile.endsWith('.html') || rootFile.endsWith('.js')) && rootFile !== 'fix_all.js') {
        let content = fs.readFileSync(rootFile, 'utf8');
        let originalContent = content;
        
        // Strategy: We will replace occurrences of 'canecas site/...' and 'Placas de MDF/...' 
        // with their lowercased, hyphenated versions.
        
        // Replace folder names first in case there are partial matches, but wait,
        // it's safer to use the exact mapping for full paths.
        
        // Find all strings like "canecas site/something.jpg" or 'canecas site/...'
        const regex = /(canecas site|Placas de MDF)\/([^"'\\]+)/gi;
        content = content.replace(regex, (match, dir, file) => {
            const oldPathLower = match.toLowerCase();
            if (mapOldToNew[oldPathLower]) {
                return mapOldToNew[oldPathLower];
            } else {
                // If it wasn't strictly found in the mapping, try to clean it manually
                const cleanDir = dir.toLowerCase() === 'canecas site' ? 'canecas-site' : 'placas-de-mdf';
                return `${cleanDir}/${cleanFilename(file)}`;
            }
        });

        if (content !== originalContent) {
            fs.writeFileSync(rootFile, content, 'utf8');
            console.log(`Updated paths in ${rootFile}`);
        }
    }
}

console.log("Fix completed successfully.");

File: fix_all.py
import os
import re

def slugify(text):
    text = text.lower()
    text = text.replace('%20', '-')
    text = re.sub(r'\s+', '-', text)
    text = re.sub(r'-+', '-', text)
    return text.strip('-')

def clean_filename(filename):
    name, ext = os.path.splitext(filename)
    return slugify(name) + ext.lower()

dirs_to_rename = {
    'canecas site': 'canecas-site',
    'canecas-site': 'canecas-site', 
    'Placas de MDF': 'placas-de-mdf',
    'placas-de-mdf': 'placas-de-mdf'
}

map_old_to_new = {}

for old_dir, new_dir in dirs_to_rename.items():
    if os.path.exists(old_dir) and old_dir != new_dir:
        os.rename(old_dir, new_dir)
        
    if os.path.exists(new_dir):
        for filename in os.listdir(new_dir):
            old_filepath = os.path.join(new_dir, filename)
            if os.path.isfile(old_filepath):
                new_filename = clean_filename(filename)
                if new_filename != filename:
                    new_filepath = os.path.join(new_dir, new_filename)
                    if os.path.exists(new_filepath):
                        base, ext = os.path.splitext(new_filename)
                        new_filename = f"{base}-1{ext}"
                        new_filepath = os.path.join(new_dir, new_filename)
                    os.rename(old_filepath, new_filepath)
                
                old_variants = [old_dir, 'canecas site', 'Placas de MDF', 'canecas-site', 'placas-de-mdf']
                for ov in old_variants:
                    old_url_path = f"{ov}/{filename}".lower()
                    new_url_path = f"{new_dir}/{new_filename}"
                    map_old_to_new[old_url_path] = new_url_path
                    # Also map the uncleaned original filename if it had spaces/caps
                    old_url_path2 = f"{ov}/{new_filename}".lower()
                    map_old_to_new[old_url_path2] = new_url_path

def replace_match(match):
    dir_part = match.group(1)
    file_part = match.group(2)
    old_full = f"{dir_part}/{file_part}".lower()
    
    if old_full in map_old_to_new:
        return map_old_to_new[old_full]
    else:
        clean_dir = "canecas-site" if "caneca" in dir_part.lower() else "placas-de-mdf"
        return f"{clean_dir}/{clean_filename(file_part)}"

regex = re.compile(r'(canecas site|Placas de MDF|canecas-site|placas-de-mdf)/([^"\'\\]+)', re.IGNORECASE)

for root_file in os.listdir('.'):
    if (root_file.endswith('.html') or root_file.endswith('.js')) and root_file not in ['fix_all.js', 'fix_all.py']:
        content = None
        enc = 'utf-8'
        try:
            with open(root_file, 'r', encoding='utf-8') as f:
                content = f.read()
        except UnicodeDecodeError:
            enc = 'latin1'
            with open(root_file, 'r', encoding='latin1') as f:
                content = f.read()
            
        if content is not None:
            original_content = content
            content = regex.sub(replace_match, content)
            
            if content != original_content:
                with open(root_file, 'w', encoding=enc) as f:
                    f.write(content)
                print(f'Updated paths in {root_file}')

print("Fix completed successfully.")
